.orchestra/support/revng-container/Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM docker.io/library/ubuntu:22.04 as BASE
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+COPY apt.no-extra /etc/apt/apt.conf.d/no-extra
+RUN apt update && \
+    apt install -y \
+        # Needed for recompilation pipe (-dev included due to linking problems)
+        libc6 libc6-dev \
+        # Needed for revng.cli etc.
+        python3 ca-certificates \
+        # some LLVM libs are not searched properly
+        libllvm12 libc++1-12 libatomic1 && \
+    apt upgrade -y && \
+    rm -rf /var/lib/apt/lists/* /var/log/*
+
+# For some reason c++ does not look for librt.so.1 but only for librt.so
+RUN cd /lib/x86_64-linux-gnu && \
+    for file in *.so.*; do \
+        SO_FILE=$(echo $file | grep -Po '^.*\.so'); \
+        if [ ! -e $SO_FILE ]; then ln -s $file $SO_FILE; fi \
+    done
+
+
+# Runtime Phase
+FROM BASE AS REVNG_PYPKG
+
+COPY requirements.txt .
+RUN apt update && \
+    apt install -y python3-pip && \
+    pip3 install -r requirements.txt && \
+    rm -rf /var/lib/apt/lists/*
+
+
+FROM BASE AS REVNG_RUN
```

Comment:
Is all of this just not to install `python3-pip` in the final image? If so, document it.
Is it actually of significant size?

Reply:
1. Ok
2. There are 2 considerations:
  * Doing it this way guarantees that there are no temporary/leftover files from `pip`
  * From a security standpoint it's better to have less tools in the image, just in case

Reply:
2. I think it's a bit overkill, unless you foresee we might add more stuff (which we shouldn't, according to orchestra design). Also consider that once we'll move all our python dependencies *within* orchestra, this will be no longer necessary.
  For security considerations, there's https://bootstrap.pypa.io/get-pip.py. But in general, unless you set `noexec` the file system, you're not adding much protection.

Do what you think is more appropriate, considering that this will vanish once we push Python packages within orchestra.
I'm fine with both approaches given this outlook.

Reply:
Since we're going to get packages in orchestra soon-ish I've opted to not have an intermediate container and to `pip install` directly in the final container